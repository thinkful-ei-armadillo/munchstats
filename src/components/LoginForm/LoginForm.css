.LoginForm{
  text-align: center;
}

.LoginForm footer {
  padding-top:10px;
}

h2{
  text-align: center;
}