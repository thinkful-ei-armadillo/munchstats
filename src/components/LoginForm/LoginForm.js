import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Input, Label } from '../Form/Form';
import AuthApiService from '../../services/auth-api-service';
import UserContext from '../../contexts/UserContext';
import Button from '../Button/Button';
import Loading from '../Loading/Loading';
import TokenService from '../../services/token-service'
import EventsApiService from '../../services/events-api-service'
import './LoginForm.css';

class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => { }
  };

  static contextType = UserContext;

  state = { error: null };

  firstInput = React.createRef();

  handleSubmit = ev => {
    ev.preventDefault();
    this.context.loadingTrue();
    const { username, password } = ev.target;

    this.setState({ error: null });

    AuthApiService.postLogin({
      username: username.value,
      password: password.value
    })
      .then(res => {
        username.value = '';
        password.value = '';
        this.context.processLogin(res.authToken);
        console.log(TokenService.parseJwt(res.authToken));
        this.props.onLoginSuccess();
        this.context.loadingFalse();
        EventsApiService.getTodaysEvents()
          .then(resj => console.log(resj))
          .then(res => this.context.setTodayEvents(res))
          .catch(e => this.context.setError(e));
      })
      .catch(res => {
        this.setState({ error: res.error });
        this.context.loadingFalse();
      });
  };

  componentDidMount() {
    this.firstInput.current.focus();
  }

  render() {

    if(this.context.loading){
      return <div className="center"><Loading /></div>;
    } else {

      const { error } = this.state;
      return (
        <form
          className='LoginForm'
          onSubmit={this.handleSubmit}>
          <div role='alert'>
            {error && <p>{error}</p>}
          </div>
          <div>
            <Label htmlFor='login-username-input'>
            Username
            </Label>
            <br />
            <Input
              ref={this.firstInput}
              id='login-username-input'
              name='username'
              required
            />
          </div>
          <br />
          <div>
            <Label htmlFor='login-password-input'>
            Password
            </Label>
            <br />
            <Input
              id='login-password-input'
              name='password'
              type='password'
              required
            />
          </div>
          <br />
          <Button type='submit'>
          Log In
          </Button>
          <footer>
            <p>Don't have an account? <Link to='/register' style={{textDecoration: 'none'}}>Sign Up!</Link></p>
          </footer>
        </form>
      );
    }}
}

export default LoginForm;