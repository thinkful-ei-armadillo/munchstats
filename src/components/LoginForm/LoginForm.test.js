import React from 'react';
import LoginForm from './LoginForm';
import {MemoryRouter} from 'react-router-dom';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<LoginForm />', () => {
  it('Renders without crashing', () => {
    mount(<MemoryRouter><LoginForm /></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><LoginForm /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});