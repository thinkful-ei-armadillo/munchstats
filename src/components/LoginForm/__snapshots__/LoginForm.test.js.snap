// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div>
    <label
      className="Label"
      htmlFor="login-username-input"
    >
      Username
    </label>
    <br />
    <input
      className="Input"
      id="login-username-input"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <br />
  <div>
    <label
      className="Label"
      htmlFor="login-password-input"
    >
      Password
    </label>
    <br />
    <input
      className="Input"
      id="login-password-input"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <br />
  <button
    className="Button"
    type="submit"
  >
    Log In
  </button>
  <footer>
    <p>
      Don't have an account? 
      <a
        href="/register"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Sign Up!
      </a>
    </p>
  </footer>
</form>
`;
